File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int out = 0;
        stack<int> stk;
        string operators = "+-*/";
        for (string str : tokens)
        {
            if(operators.find(str) != string::npos)
            {
                int elem2 = stk.top(); stk.pop();
                
                int elem1 = stk.top(); stk.pop();
                
                if(str == "+")
                    stk.push(elem1 + elem2);
                else if(str == "-")
                    stk.push(elem1 - elem2);
                else if(str == "*")
                    stk.push(elem1 * elem2);
                else if(str == "/")
                    stk.push(elem1 / elem2);
            }
            else
                stk.push(stoi(str));
        }
        return stk.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stk = []
        operators = "+-*/"
        for elem in tokens:
            if elem not in operators:
                stk.append(int(elem))
            else:
                first = stk.pop()
                second = stk.pop()
                if elem == "+":
                    stk.append(first + second)
                elif elem == "-":
                    stk.append(second - first)
                elif elem == "*":
                    stk.append(first * second)
                else:
                    stk.append(int(second / first))
        return stk[-1]
